### DISCUSSION TITLE
Why does sqlmodel define `class_registry` at the top level of its main.py?

### DISCUSSION THREAD

**Discussion by zmievsa** [fastapi/sqlmodel#1462] [Questions]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the SQLModel documentation, with the integrated search.
- [X] I already searched in Google "How to X in SQLModel" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to SQLModel but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [x] I already checked if it is not related to SQLModel but to [SQLAlchemy](https://github.com/sqlalchemy/sqlalchemy).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from sqlmodel import class_registry
```

### Description

Why does sqlmodel define `class_registry` at the top level of its main.py? It seems to be related to SQLAlchemy but it does not seem to be used anywhere.

### Operating System

Linux, Windows, macOS, Other

### SQLModel Version

0.0.8

### Python Version

3.11

**Comment by FaridFArab** [comment]
Hi @zmievsa, I think your question be answered  [here](https://github.com/tiangolo/sqlmodel/issues/245)

**Comment by theunkn0wn1** [comment]
Hi, @FaridFArab , i looked at your linked discussion but it offers no answers.

**Comment by FaridFArab** [comment]
OK I will answer here;
when you use class_registry, it serves as a mapping that stores all the models defined in your application. This registry keeps track of the models and helps SQLModel manage relationships and queries between these models effectively.
Here is a step-by-step explanation of the purpose of class_registry in SQLModel:
1. Model Registration: When you define a model class in SQLModel, it needs to be registered so that SQLModel can work with it efficiently. The class_registry is where these model classes are stored after being defined.
2. Model Access and Relationship Handling: SQLModel uses the class_registry to access the model classes during query generation and execution. This is essential for handling relationships between models and constructing queries that involve multiple models.
3. Query Execution: When you perform CRUD operations or queries using SQLModel, the framework relies on the class_registry to fetch the relevant model classes and perform the necessary operations.
4. Dynamic Model Handling: The class_registry allows SQLModel to work with models dynamically. This means that you can define models at runtime and still have SQLModel manage them effectively through this registry.
By centralizing the storage of model classes in the class_registry, SQLModel can streamline how models are accessed, queried, and related to each other within your application.
I hope my answer will be useful for you.
